src/Controls/tests/DeviceTests/Elements/Layout/LayoutTests.cs

```diff
@@ -236,5 +236,46 @@ async Task FlexLayoutInStackLayoutDoesNotCycle(IStackLayout root)
 				await AttachAndRun(root, (handler) => { });
 			});
 		}
+
+		[Fact]
+		public async Task GridCellsHonorMaxWidth()
+		{
+			var grid = new Grid() { MaximumWidthRequest = 50 };
+			var label = new Label() { Text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sodales eros nec massa facilisis venenatis", LineBreakMode = LineBreakMode.WordWrap};
+
+			grid.Add(label);
+
+			await InvokeOnMainThreadAsync(async () =>
+			{
+				await CreateHandlerAsync<LabelHandler>(label);
+				await CreateHandlerAsync<LayoutHandler>(grid);
+
+				await AttachAndRun(grid, (handler) => { });
+			});
+
+			Assert.True(label.Width <= grid.MaximumWidthRequest);
+			Assert.True(grid.Width <= grid.MaximumWidthRequest);
+		}
+
+		[Fact]
+		public async Task GridCellsHonorMaxHeight()
+		{
+			var grid = new Grid() { MaximumWidthRequest = 20 };
+			var label = new Label() { Text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sodales eros nec massa facilisis venenatis", LineBreakMode = LineBreakMode.WordWrap };
+
+			grid.Add(label);
+
+			await InvokeOnMainThreadAsync(async () =>
+			{
+				await CreateHandlerAsync<LabelHandler>(label);
+				await CreateHandlerAsync<LayoutHandler>(grid);
+
+				await AttachAndRun(grid, (handler) => { });
+			});
+
+			Assert.True(label.Width <= grid.MaximumHeightRequest);
```

Comment:
Looks like these are comparing the wrong values?

Reply:
Doh. Sloppy copy paste